src/Compilers/Test/Core/Assert/AssertEx.cs

```diff
@@ -1013,5 +1023,75 @@ public static void Contains<T>(IEnumerable<T> collection, Predicate<T> filter, F
             Fail("Filter does not match any item in the collection: " + Environment.NewLine +
                 ToString(collection, itemSeparator ?? Environment.NewLine, itemInspector));
         }
+
+#nullable enable
+
+        /// <summary>
+        /// The xunit Assert.Equal method is not callable in Visual Basic  due to the presence of
+        /// the unmanaged constraint. Need to indirect through C# here until we resolve this.
```

Comment:
I didn't understand how that works. If we're forwarding to an API that has an `unmanaged` constraint, shouldn't we have an error here (`unmanaged` constraint not honored)?

Reply:
The `unmanaged` constraint is used in xUnit as a fast path optimization. It's just allowing them to call `Span.SequenceEquals` helpers. It's not actually a functional difference. This is just a bridge that lets us still make the right calls in VB.